### ISSUE TITLE
GC_Regressions._v2_0_rtm_544701_544701_544701 times out occasionally in CI

### ISSUE THREAD

**Issue by adityamandaleeka** [dotnet/coreclr#10531]
Examples:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_windows_nt_prtest/4585/testReport/junit/GC_Regressions/_v2_0_rtm_544701_544701_544701_/_v2_0_rtm_544701_544701_544701_cmd/

https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_windows_nt_prtest/4681/testReport/junit/GC_Regressions/_v2_0_rtm_544701_544701_544701_/_v2_0_rtm_544701_544701_544701_cmd/

**Comment by adityamandaleeka** [comment]
I just looked at this test and it just allocates byte arrays in a loop, starting at 41938869 bytes and incrementing the number of bytes all the way to 41943020. Looking at those logs, it really does appear to be a normal time-out (nothing nefarious is going on) and we simply don't get a chance to get all the way to the end of the loop.

@jkotas @Maoni0 I'm not sure what the context of this test is (presumably there was some bug in the distant past that this test addresses), but if it is really taking a long time on our CI jobs, can we just have it increase the byte count by, say, 7 each time instead of incrementing? I'd rather not increase the timeout for something that seems kind of low-value.

**Comment by Maoni0** [comment]
did this test just get added to the CI run? has this timeout always been there?

**Comment by adityamandaleeka** [comment]
I don't know how far back it goes, but in the last ~150 jobs for Windows x64 Debug it has occurred 5 times. I don't think we have CI logs further back than that, but we know it has occurred even before then since @janvorli also reported it 21 days ago: https://github.com/dotnet/coreclr/issues/3939#issuecomment-284711299

**Comment by Maoni0** [comment]
The worry is if this is a new issue - assuming we had not reduced the time out value, if this timeout didn't happen before but started to happen at some point that would point to a perf problem (it is a debug build but this doesn't use much other than the GC). Would it be possible to run it with a dbg build that's from a while back (eg a 1.0 build)?

**Comment by adityamandaleeka** [comment]
I'm running a little experiment in CI to see (a) how long this test is taking in CI in general and (b) whether it has regressed at all compared to 2 months ago.

Locally it runs in about 6 seconds on my (fairly powerful) machine. The default timeout in CI appears to be 10 minutes, so the test is running at least 100x slower on CI when these failures occur. I wonder if the excessive WriteLines have anything to do with it...

**Comment by adityamandaleeka** [comment]
@Maoni0 Nice, we posted at about the same time 😄. I was worried about the same thing, which is why I'm running the experiment. I'll report back with results.

**Comment by adityamandaleeka** [comment]
Unfortunately, the regression part of the experiment wasn't successful because of some other changes that have occurred in the past few weeks (the CI is passing in arguments that the test scripts didn't know about before, which is causing failures). That will require a little more work to get working.

However, I did find that the test is generally taking a long time in CI. Here's how long it took in my 4 trials:

```
8:39
8:38
8:46
9:48
```

With this as the normal case, it's not unreasonable that sometimes we're hitting the 10 minute limit.

I've started another set of runs to test my theory about the excessive writes to stdout.

**Comment by adityamandaleeka** [comment]
Yup, as suspected, all the `Console.Write` calls are slowing things down in CI (the test infrastructure writes the output to a file too). I removed most of the console writing and ran 4 more times, with the following durations:

```
4718  3:18
4719  4:53
4720  1:50
4721  1:28
```

On a related note, I also noticed that the chaos65204782cs tests in JIT/Generics have timed out a few times as well. Those tests also write a ton of stuff to the console.

**Comment by Maoni0** [comment]
great! 

it seems odd that there's so much variation (~1.5 mins to ~5mins). I presume the time is measured by running this test on one thread in the test infra and with no other tests running at the same time?

**Comment by adityamandaleeka** [comment]
No, I just let the CI operate normally (since we wanted to know how long it's taking in general), so other tests _are_ running in parallel. Since the timing of things can affect exactly what else is running at the same time as this test, the execution times have a large variance.

**Comment by adityamandaleeka** [comment]
As a final experiment, I ran this through CI 4 more times without any other tests running and it took between 4 and 6 seconds in every case. I'm going to go ahead and check in a change to this test that reduces the number of Console.Write*s as I did above, so we shouldn't be hitting this timeout anymore under normal circumstances.

The fact that writing stuff to stdout when tests are running in parallel is as expensive as it is is something that should be inspected in the test infrastructure. There must be something they can do differently in terms of how the output is being handled. If I have some spare time, I'll take a look and make a recommendation.
